cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -343,4 +353,18 @@ private ProfileId getResolvedProfileId(EntityId entityId) throws IOException {
     }
     return programIds;
   }
+
+  /**
+   * Emit the application count metric.
+   */
+  private void emitApplicationCountMetric() {
+    long pipelineCount = 0;
+    try {
+      pipelineCount = appMetadataStore.getApplicationCount();
+    } catch (IOException e) {
+      LOG.warn("Failed to get application count", e);
+    }
+    metricsCollectionService.getContext(Collections.emptyMap()).gauge(Constants.Metrics.Program.APPLICATION_COUNT,
```

Comment:
Move this inside the `try` so that on exception, the count won't get updated to 0.

Reply:
done